### PR TITLE
refactor: use subcollection to store user songs

### PR DESCRIPTION
**Pull request by sriramb2000**
Title. Also added a button in the UI to post song, using your currently playing song, along with your current location. Geolocation API requires HTTPS, so configured start script to use HTTPS.

image

### CODE REVIEW

File: frontend/web-2-fac-auth-demo/src/App.js [hunk 1]
```diff
@@ -157,6 +157,33 @@ function App() {
       });
   }
 
+  function postSong() {
+    if (!currentlyPlaying) {
+      // Should probably just hit the endpoint ourselves tbh but too lazy.
+      return window.alert("Cannot post song without getting your current song data fuckwad...");
```
Comment (michaelw54) [inline frontend/web-2-fac-auth-demo/src/App.js:163]: I think you might want to remove "fuckwad"

File: frontend/web-2-fac-auth-demo/src/App.js [hunk 2]
```diff
@@ -157,6 +157,33 @@ function App() {
...
+            two_fac_token: twoFacToken,
+          };
+          axios.post(`${BASE_URL}/api/songs`, {
+            lat,
+            lng,
+            songData
+          }, { headers: authHeaders }).then((res) => window.alert(`Posted successfully: ${JSON.stringify(res.data)}`));
+        });
+    }, (err) => {
+      window.alert("Cannot post song without location data fuckwad...");
```
Comment (michaelw54) [inline frontend/web-2-fac-auth-demo/src/App.js:183]: Bro stop with the "fuckwad"

File: backend/src/routes/songs.js [hunk 3]
```diff
@@ -10,19 +10,22 @@ async function handleSong(req, res) {
...
-  const result = await db.collection('songEntries').add({
+  const songDoc = {
     songData,
     locationHash: geofire.geohashForLocation([lat, lng]),
     lat,
     lng,
     ts: firebase.firestore.FieldValue.serverTimestamp(),
-  });
+  };
+  const result = await db.collection('songEntries').add(songDoc);
```
Comment (michaelw54) [inline backend/src/routes/songs.js:20]: We may need to dedupe songEntries in your area if many people in the area submit the same song. Is it possible to filter unique on a field of a field (like get all unique songData.songId within a radius)

Comment (michaelw54) [reply backend/src/routes/songs.js:20]: @sjiang647 do you think we can add deduping logic in the /nearby endpoint to only get unique songEntries

**Review by michaelw54 — APPROVED** [review]
LGTM but remove the "fuckwad" thx

File: frontend/web-2-fac-auth-demo/src/App.js (hunk 2 above)
Comment (legitmaxwu) [reply frontend/web-2-fac-auth-demo/src/App.js:183]: LMAO

**Pull request merged**